tox.ini

```diff
@@ -128,7 +128,7 @@ pass_env =
     CI_PACKED_CHARMS
 commands_pre =
     poetry install --with integration
-    poetry run pip install juju==2.9.42.1
+    poetry run pip install juju==3.1.2.0
 commands =
     poetry run pytest -v --tb native --log-cli-level=INFO -s --durations=0 {posargs} {[vars]tests_path}/integration/test_teardown.py
```

Comment:
Again, the `tox.ini` changs are NOT part of the PR -- were only needed for the pipeline. Will NOT be merged.

([The PR that IS to be merged](https://github.com/canonical/mongodb-k8s-operator/pull/161) is equal to the current one, except the last commit.)